transaction/core/src/tx_summary.rs

```diff
@@ -0,0 +1,210 @@
+    pub fee_token_id: u64,
+
+    /// The block index at which this transaction is no longer valid.
+    #[prost(uint64, tag = "5")]
+    pub tombstone_block: u64,
+}
+
+impl TxSummary {
+    /// Make a TxSummary for a given TxPrefix and pseudo-output commitments
+    pub fn new(
+        tx_prefix: &TxPrefix,
+        pseudo_output_commitments: &[CompressedCommitment],
+    ) -> Result<Self, ZipExactError> {
+        // Scratch which helps us associate outputs to inputs with rules
+        let mut input_rules_associated_target_keys: BTreeSet<CompressedRistrettoPublic> =
+            Default::default();
+
+        // Compute the inputs
+        let inputs: Vec<TxInSummary> =
+            zip_exact(tx_prefix.inputs.iter(), pseudo_output_commitments.iter())?
+                .map(|(input, commitment)| {
+                    let mut result = TxInSummary {
+                        pseudo_output_commitment: commitment.clone(),
+                        ..Default::default()
+                    };
+                    if let Some(rules) = &input.input_rules {
+                        result.has_input_rules = true;
+                        let mut associated_keys = rules.associated_tx_target_keys();
+                        input_rules_associated_target_keys.append(&mut associated_keys);
+                    }
+                    result
+                })
+                .collect();
+
+        let outputs: Vec<TxOutSummary> = tx_prefix
+            .outputs
+            .iter()
+            .map(|src| TxOutSummary {
+                masked_amount: src.masked_amount.clone(),
+                target_key: src.target_key,
+                public_key: src.public_key,
+                associated_to_input_rules: input_rules_associated_target_keys
+                    .contains(&src.target_key),
+            })
+            .collect();
+
+        Ok(Self {
+            outputs,
+            inputs,
+            fee: tx_prefix.fee,
+            fee_token_id: tx_prefix.fee_token_id,
+            tombstone_block: tx_prefix.tombstone_block,
+        })
+    }
+}
+
+/// A subset of the data of a TxOut.
+///
+/// Fog hint and memo are omitted to reduce size and complexity on HW device,
+/// which can't really do much with those and isn't very interested in them
+/// anyways.
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
+pub struct TxOutSummary {
+    /// The amount being sent.
+    #[prost(oneof = "MaskedAmount", tags = "1, 2")]
+    pub masked_amount: Option<MaskedAmount>,
+
+    /// The one-time public address of this output.
+    #[prost(message, required, tag = "3")]
+    pub target_key: CompressedRistrettoPublic,
+
+    /// The per output tx public key
+    #[prost(message, required, tag = "4")]
+    pub public_key: CompressedRistrettoPublic,
+
+    /// Whether or not this output is associated to an input with rules
+    #[prost(bool, tag = "5")]
+    pub associated_to_input_rules: bool,
```

Comment:
:question: Why is the input rules flag here?  If the client wants to ensure input rules shouldn't it know the rules themselves?

Maybe hw wallets won't used with SCI, but if the customer wants to swap 100mob and get back 100eusd, the application could change this to swap 100mob for 10eusd and nothing would say differently here.

Reply:
can't easily explain right now, but the goal is basically to reduce the bandwidth to the device and not send all of the input rules, which may be quite big. The device doesn't actually need to see the rules to meet its requirements, the only reason it needs this flag is so that it can know that it shouldn't expect to see a public address for this output.

Reply:
See comment below, regarding what is the threat model.
I'm assuming that the HW wallet is the trusted device and that the application it's talking to isn't trustworthy.
If the application is fully trustworthy then it's probably ok, but if not, I don't have enough understanding to say one way or another, but it feels like this is leaving out sufficient information for the HW wallet.